wagtail/admin/tests/test_pages_views.py

```diff
@@ -361,25 +361,21 @@ def test_admin_at_site_page(self):
     def test_nonadmin_at_root(self):
         self.assertTrue(self.client.login(username='siteeditor', password='password'))
         response = self.client.get(reverse('wagtailadmin_explore_root'))
-        self.assertEqual(response.status_code, 200)
-        # Non-admin should get a simple "create pages as children of the homepage" prompt
-        self.assertContains(
-            response,
-            "Pages created here will not be accessible at any URL. "
-            "To add pages to an existing site, create them as children of the homepage."
+
+        # Non-admins should be redirected to their explorable root.
+        self.assertEqual(
+            (response.status_code, response['Location']),
+            (302, reverse('wagtailadmin_explore', args=(self.site_page.pk, )))
         )
 
     def test_nonadmin_at_non_site_page(self):
         self.assertTrue(self.client.login(username='siteeditor', password='password'))
         response = self.client.get(reverse('wagtailadmin_explore', args=(self.no_site_page.id, )))
-        self.assertEqual(response.status_code, 200)
-        # Non-admin should get a warning about unroutable pages
-        self.assertContains(
-            response,
-            (
-                "There is no site record for this location. "
-                "Pages created here will not be accessible at any URL."
-            )
+
+        # Non-admins should be redirected to their explorable root.
+        self.assertEqual(
+            (response.status_code, response['Location']),
+            (302, reverse('wagtailadmin_explore', args=(self.site_page.pk, )))
         )
```

Comment:
As it stands, these tests are wrongly conflating "non-admin" with "has no permission for this view" - it's possible for a non-admin user to legitimately have access to the root section here (either by having that permission explicitly assigned at the group level, or by having permission over several deeper sections that have no common ancestor other than the root). The old version of these tests was at fault here, by failing to set up the appropriate permissions for the non-admin user - which, of course, didn't matter until we tightened up the permission checks here. There are actually three cases we need to test here, for both the root and non-root views:

* No permission for this view: user should be redirected away
* Ordinary user with permission for this view: should see the listing, with a warning message that pages created here won't be accessible
* Admin (or more precisely, any user with 'add site' permission): should see the listing, with a warning message that pages created here won't be accessible including advice about configuring a site record.